src/AutoRest.CSharp/Common/Output/Models/ClientMethodBodyLines.cs

```diff
@@ -0,0 +1,216 @@
+                        arguments.Add(CreateConversion(new VariableReference(variable), model.Type, protocolParameter.Type));
+                        break;
+                    case { ProtocolParameters.Count: > 1, ConvenienceParameters.Count: 1, IntermediateModel: {} model }:
+                        // Grouping is not supported yet
+                        break;
+                }
+            }
+
+            protocolMethodArguments = arguments;
+        }
+
+        private static ValueExpression CreateConversion(Parameter fromParameter, CSharpType toType)
+        {
+            return fromParameter.Type switch
+            {
+                { IsFrameworkType: false, Implementation: EnumType { IsExtensible: true } }  when toType.EqualsIgnoreNullable(typeof(string)) => Call.ToString(fromParameter),
+                { IsFrameworkType: false, Implementation: EnumType { IsExtensible: false } } when toType.EqualsIgnoreNullable(typeof(string)) => Call.ToSerialString(fromParameter),
+                { IsFrameworkType: false, Implementation: ModelTypeProvider }                when toType.EqualsIgnoreNullable(typeof(RequestContent)) => Call.ToRequestContent(fromParameter),
+                _ => new ParameterReference(fromParameter)
+            };
+        }
+
+        private static ValueExpression CreateConversion(ValueExpression fromExpression, CSharpType fromType, CSharpType toType)
+        {
+            if (fromType.IsNullable && toType.IsNullable)
+            {
+                fromExpression = new NullConditionalExpression(fromExpression);
+            }
+
+            return fromType switch
+            {
+                { IsFrameworkType: false, Implementation: EnumType { IsExtensible: true } }  when toType.EqualsIgnoreNullable(typeof(string)) => Call.ToString(fromExpression),
+                { IsFrameworkType: false, Implementation: EnumType { IsExtensible: false } } when toType.EqualsIgnoreNullable(typeof(string)) => Call.ToSerialString(fromType, fromExpression),
+                { IsFrameworkType: false, Implementation: ModelTypeProvider }                when toType.EqualsIgnoreNullable(typeof(RequestContent)) => Call.ToRequestContent(fromExpression),
+                _ => fromExpression
+            };
+        }
+
+        public static class Assign
+        {
+            public static MethodBodyLine ResponseValueId(CodeWriterDeclaration response, ValueExpression from)
+            {
+                return new SetValueLine(GetResponseValueId(response), from);
+            }
+        }
+
+        public static class Declare
+        {
+            public static MethodBodyLine FirstPageRequest(ValueExpression? restClient, string methodName, IEnumerable<ValueExpression> arguments, out CodeWriterDeclaration localFunctionName)
+            {
+                var requestMethodCall = new InstanceMethodCallExpression(restClient, methodName, arguments.ToList(), false);
```

Comment:
If we go this route it starts to get harder to read the code and determine what the code will look like in the end.

I wonder if we should consider something along the lines of what exists with [ModelMethodDefinition](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Types/ModelMethodDefinition.cs).

The idea here is that each writer can create simple templated methods like [FromRequest and ToResponseContent](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Generation/Writers/SerializationWriter.cs#L24-L25) methods and then the output types can [combine](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs#L393) the signature with the body implementation they want.

This maintains a fairly easy visual translation between the generator code and the resulting dotnet code for the project.  Yes there will be some duplication around some minor things but as we find significant ones we can abstract those out so they can get called.

At a minimum we shouldn't have two ways to do this since this is a consolidation effort.